File: 프로그래머스/unrated/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    answer = []
    HOF = []
    for s in score:
        HOF.append(s)
        if len(HOF) < k:
            answer.append(min(HOF))
        else:
            HOF.sort(reverse=True)
            answer.append(min(HOF[:k]))
            
    return answer

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.js
function solution(k, score) {
    var answer = [];
    const HOF = []
    
    for (let i = 0; i < score.length; i++) {
        HOF.push(score[i])
        HOF.sort((a, b) => b - a)
        if (HOF.length === k + 1) HOF.pop()
        answer.push(HOF.at(-1))        
        
    }
    return answer;
}